File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/java/io/github/lishangbu/avalon/dataset/service/StatService.java
package io.github.lishangbu.avalon.dataset.service;

import io.github.lishangbu.avalon.dataset.entity.Stat;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/// 能力(Stat)服务。
public interface StatService {

    /// 根据条件分页查询能力。
    Page<Stat> getPageByCondition(Stat stat, Pageable pageable);

    /// 新增能力。
    Stat save(Stat stat);

    /// 更新能力。
    Stat update(Stat stat);

    /// 根据主键删除能力。
    void removeById(Long id);

    /// 根据条件查询能力列表。
    List<Stat> listByCondition(Stat stat);
}

File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/kotlin/io/github/lishangbu/avalon/dataset/service/StatService.kt
package io.github.lishangbu.avalon.dataset.service

import io.github.lishangbu.avalon.dataset.entity.dto.SaveStatInput
import io.github.lishangbu.avalon.dataset.entity.dto.StatSpecification
import io.github.lishangbu.avalon.dataset.entity.dto.StatView
import io.github.lishangbu.avalon.dataset.entity.dto.UpdateStatInput

/** 能力值服务*/
interface StatService {
    /** 保存能力值*/
    fun save(command: SaveStatInput): StatView

    /** 更新能力值*/
    fun update(command: UpdateStatInput): StatView

    /** 按 ID 删除能力值*/
    fun removeById(id: Long)

    /** 根据条件查询能力值列表*/
    fun listByCondition(specification: StatSpecification): List<StatView>
}
